From bcc9a014e94c4bdc4ff575468278dc9333fa0d1b Mon Sep 17 00:00:00 2001
From: Joel Klussman <joel.klussman@intel.com>
Date: Mon, 25 Feb 2013 15:14:40 -0800
Subject: [PORT FROM MAIN] wifi: Wifi hotspot enhancements

BZ: 86323

DO NOT UPSTREAM

Wifi hotspot enhancements - move hotspot to top level settings,
                            add hotspot widget,
                            show connected devices

This is a Mike Bell feature request.
Backported from PPD as is.

Feature   : Wlan
Category  : Differentiation
Purpose   : AOSP-ext
Origin    : Upstream
Upstream-candidate: no

Change-Id: I15854504690f434b175f2fa46cf533da4a531e86
Orig-Change-Id: Ifc01b24ee9a3e3e3b155d5c775cfd3e4b85b3493
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 TetherController.cpp |   12 +++++++-----
 1 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/TetherController.cpp b/TetherController.cpp
index a3bb1b7..7cdf919 100644
--- a/TetherController.cpp
+++ b/TetherController.cpp
@@ -163,7 +163,8 @@ int TetherController::startTethering(int num_addrs, struct in_addr* addrs) {
             close(pipefd[0]);
         }
 
-        int num_processed_args = 9 + mInterfaces->size() + (num_addrs/2) + 1; // 1 null for termination
+        /* Wifi_Hotspot : dhcp-script is enabled */
+        int num_processed_args = 10 + mInterfaces->size() + (num_addrs/2) + 1; // 1 null for termination
         char **args = (char **)malloc(sizeof(char *) * num_processed_args);
         args[num_processed_args - 1] = NULL;
         args[0] = (char *)"/system/bin/dnsmasq";
@@ -173,11 +174,12 @@ int TetherController::startTethering(int num_addrs, struct in_addr* addrs) {
         // TODO: pipe through metered status from ConnService
         args[4] = (char *)"--dhcp-option-force=43,ANDROID_METERED";
         args[5] = (char *)"--pid-file";
-        args[6] = (char *)"-z";
-        args[7] = (char *)"-Ilo";
-        args[8] = (char *)"";
+        args[6] = (char *)"--dhcp-script=/system/bin/dhcp_lease_evt.sh";
+        args[7] = (char *)"-z";
+        args[8] = (char *)"-Ilo";
+        args[9] = (char *)"";
 
-        int nextArg = 9;
+        int nextArg = 10;
 
          /*Activate the DHCP server only on tethered interfaces*/
         InterfaceCollection *ilist = mInterfaces;
-- 
1.7.4.1

